src/app/shared/guards/pending-changes-guard.ts

```diff
@@ -44,16 +46,26 @@ export class PendingChangesGuard implements CanDeactivate<PendingChangesComponen
       icon: 'pi pi-exclamation-triangle',
       acceptLabel: $localize`Yes, leave page`,
       accept: () => {
-        dialogResponse.next(true);
-        dialogResponse.complete();
+        dialogResponse.next('accepted');
       },
       rejectLabel: $localize`No`,
       reject: () => {
-        dialogResponse.next(false);
-        dialogResponse.complete();
+        dialogResponse.next('declined');
       },
     });
-    return dialogResponse;
+
+    dialogResponse$.pipe(
+      filter(status => status !== 'pending'),
+      take(1),
+      delay(0),
+    ).subscribe(() =>
+      dialogResponse.next('pending')
+    );
+
+    return dialogResponse$.pipe(
+      filter(status => status !== 'pending'),
+      map(status => status === 'accepted'),
+    );
```

Comment:
I don't even see what you need the pending state actually as you never output it... ah that's because you have done a BehaviorSubject which keeps and re-emits its last value... but a normal subject (correctly used) should  have been enough.